File: app/src/main/java/top/maweihao/weather/entity/BDLocateApi.java
package top.maweihao.weather.entity;

import io.reactivex.Observable;

import retrofit2.http.GET;
import retrofit2.http.Query;
import top.maweihao.weather.entity.BaiDu.BaiDuChoosePositionBean;
import top.maweihao.weather.entity.BaiDu.BaiDuCoordinateBean;
import top.maweihao.weather.entity.BaiDu.BaiDuIPLocationBean;

/**
 * interface for retrofit
 * Created by maweihao on 26/11/2017.
 */

public interface BDLocateApi {

    @GET("location/ip")
    Observable<BaiDuIPLocationBean> getIpLocation(@Query("ak") String ak,
                                                  @Query("coor") String coor,
                                                  @Query("mcode") String mcode);

    @GET("geocoder/v2/")
    Observable<BaiDuCoordinateBean> getAddressDetail(@Query("location") String location,
                                                     @Query("output") String output,
                                                     @Query("pois") int pois,
                                                     @Query("ak") String ak,
                                                     @Query("mcode") String mcode);

    @GET("geocoder/v2/")
    Observable<BaiDuChoosePositionBean> getCoordinateByDesc(@Query("address") String location,
                                                            @Query("output") String output,
                                                            @Query("ak") String ak,
                                                            @Query("mcode") String mcode);
}

File: app/src/main/java/top/maweihao/weather/util/http/api/BDLocateApi.kt
package top.maweihao.weather.util.http.api

import io.reactivex.Flowable
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST
import top.maweihao.weather.entity.BaiDu.BDIPLocationBean
import top.maweihao.weather.entity.BaiDu.BaiDuChoosePositionBean
import top.maweihao.weather.entity.BaiDu.BaiDuCoordinateBean

//百度定位API
interface BDLocateApi {

    @FormUrlEncoded
    @POST("location/ip")
    fun getIpLocation(@Field("ak") ak: String,
                      @Field("coor") coor: String,
                      @Field("mcode", encoded = false) mcode: String): Flowable<BDIPLocationBean>

    @FormUrlEncoded
    @POST("geocoder/v2/")
    fun getAddressDetail(@Field("location") location: String,
                         @Field("output") output: String,
                         @Field("pois") pois: Int,
                         @Field("ak") ak: String,
                         @Field("mcode") mcode: String): Flowable<BaiDuCoordinateBean>

    @FormUrlEncoded
    @POST("geocoder/v2/")
    fun getCoordinateByDesc(@Field("address") location: String,
                            @Field("output") output: String,
                            @Field("ak") ak: String,
                            @Field("mcode") mcode: String): Flowable<BaiDuChoosePositionBean>
}